pkg/runner/expression.go

```diff
@@ -499,7 +499,7 @@ func getEvaluatorInputs(ctx context.Context, rc *RunContext, step step, ghc *mod
 		}
 	}
 
-	if ghc.EventName == "workflow_dispatch" {
+	if rc.caller == nil && ghc.EventName == "workflow_dispatch" {
```

Comment:
Hello, in case of `workflow_dispatch` to the `main.yml` the `workflow_dispatch` values from `reuse.yml` (`workflow_call`) is not needed. I checked with a test workflow on Github.

### main.yml

```yaml
name: CI

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

  workflow_dispatch:

jobs:
  build:
    runs-on: [ self-hosted ]
    steps:
      - uses: actions/checkout@v4
      - name: Run a one-line script
        run: echo "✅ 🚀 ✅ hello this is from workflow main" ${{ github.event_name }}  
  call-reuse-w-val:
    uses: ./.github/workflows/reuse.yml
    with:
      my-val: "passed value from main"
```

### reuse.yml

```yaml
name: reuse

on:
  workflow_dispatch:
    inputs:
      my-val:
        type: string
        required: true
        default: "default_value_reuse_workflow_dispatch_call"
      dispatch-val:
        type: string
        default: "I am a dispatch var for checking if I am being used in workflow_call"

  workflow_call:
    inputs:
      my-val:
        type: string
        required: true
        default: "default_value_reuse_workflow_call"

jobs:
  reusable_workflow_job:
    runs-on: [ self-hosted ]
    steps:
      - uses: actions/checkout@v4
      - name: Run a one-line script
        run: echo "✅ 🚀 ✅ hello this is from workflow reuse. Value - " ${{ inputs.my-val }} ${{ github.event_name }} ${{ inputs.dispatch-val }}
```

### dispatch (main) workflow_call (reuse) output

```
✅ 🚀 ✅ hello this is from workflow main workflow_dispatch
✅ 🚀 ✅ hello this is from workflow reuse. Value -  passed value from main workflow_dispatch
```

### dispatch output (reuse)
```
✅ 🚀 ✅ hello this is from workflow reuse. Value -  default_value_reuse_workflow_dispatch_call workflow_dispatch I am a dispatch var for checking if I am being used in workflow_call
```

If I try to replace the value of `dispatch-val` from main

```
The workflow is not valid. .github/workflows/main.yml (Line: 39, Col: 21): Invalid input, dispatch-val is not defined in the referenced workflow.
```